src/server/index.ts

```diff
@@ -59,20 +113,86 @@ function startMatch(players: [User, User]) {
 		}
 	}, 1000);
 
+
 	// Start battle
-	movePlayerToBattle(players[0], players[1]);
-	movePlayerToBattle(players[1], players[0]);
+
+	if (isBot(players[1]))   
+	{
+		movePlayerToBotBattle(players[0], players[1]);
+		// movePlayerToBotBattle(players[1], players[0]);
+	}
+
+	else
+	{
+	 	movePlayerToBattle(players[0], players[1]);
+	 	movePlayerToBattle(players[1], players[0]);
+	}
+}
+
+
+function startBotMatch(player: User)
+{
+	console.log("start bot match");
+
+	pickAction(botUser, null);
+	startMatch([player, botUser]);
+}
+
+
+function isBot(p: User)    // todo: would be better to check if playerStates[0] is 'bot-battle' but that breaks it (currently, adding a client called 'bot' creates a lot of weirdness)
```

Comment:
There would never be a client who connects with id bot in production so not much to worry about unless I misunderstood the concern :)